app/tailwind.css

```diff
@@ -0,0 +1,12 @@
+@tailwind base;
+@tailwind components;
+@tailwind utilities;
+
+@layer base {
+  h1 {
+    @apply text-4xl;
+  }
+  h2 {
+    @apply text-2xl;
+  }
```

Comment:
I added them (without giving it much thought).

I have two concerns here (in order):
1. How do we achieve consistency across the UI?
2. It's annoying to have to redo basic styling on generic components like h1, h2, and p.

I definitely don't think we should set up many of these, but I do think there is benefit to the 'cascade' part of CSS. I think I'd be particularly inclined to have base styles for typography (perhaps sizing, font-weight, and font-family?). Also, since they're in the base layer they're easy to override in specific situations.

I do see what you're saying about when it's all components though...

One alternative suggestion for UI consistency could be setting up some custom classes - e.g. text-size-h1, text-size-h2, not sure about this though...

Reply:
I'll just remove them for now.

Reply:
Great questions and it's getting to the heart of some people's objections to Tailwind - it seems to be working against the ideas of CSS for styling markup in a reusable way (and using the cascade etc).

In terms of how to achieve consistency, and reuse styles this page is worth reading: https://tailwindcss.com/docs/reusing-styles

The TLDR is:
- style duplication in a single file is probably fine, especially if you use "multiple cursors" and other editor features to make updating them all easy
- for style duplication across a project, a component is often a better abstraction (in React anyway) because it lets you control other things too, like the markup

Setting some global styles that apply to all `<h2>` elements for example is tempting at this stage of the project, the place it becomes painful is:
- if the project gets big enough that you forget all the places `<h2>` is used, and make a change on one page and don't realise it's breaking a different page
- if you start using 3rd party NPM packages that use markup like `<h2>` but _should not_ have your styles applied, and your styles break the layout of those components.

This project is not likely to get _that_ big, so if you feel it's going to be better to have some basic typography styles that are applied globally, I'm okay with it. I just wanted to draw attention to the direction Tailwind encourages you to go.